### ISSUE TITLE
Installing austraits on a fresh new device - lots of dependencies resulting in error of installation

### ISSUE THREAD

**Issue by yangsophieee** [traitecoevo/austraits#47]
I'm installing the `austraits` package on a new device (Windows 10) with a fresh installation of R and RStudio. The installation seems to work fine until the creating vignettes step, where it fails if you don't have certain packages installed. So the installation would fail with an error like: ```Error: processing vignette 'austraits.Rmd' failed with diagnostics: there is no package called 'janitor'```. I would install the missing package and try again, and it would fail again because of another package. This happened for the following packages: `janitor`, `raster`, `ggpointdensity`, `viridis`, and `ggbeeswarm`. The installation of `austraits` succeeded in the end after installing all these dependencies.

In the end, it worked but I'm wondering if there is a faster way to figure out what package dependencies you are missing, instead of finding out what you are missing one by one.

**Comment by fontikar** [comment]
Hey @yangsophieee thanks for letting us know!

Yes there is a way to install ALL the dependencies for `austraits`. The packages you listed above are considered as 'soft dependencies' and by default, are not installed when you use `remotes::install_github("traitecoevo/austraits"`. This was a way to keep the R package at a smaller size but I think @dfalster perhaps we should reconsider whether we should put them back as Imports as some of the functions e.g. summarise_austraits and both plotting functions need those packages. 

To unlock the _full_ potential of austraits, I suggest calling `remotes::install_github("traitecoevo/austraits", dependencies = TRUE)`, this *should* theoretically install all the packages that austraits need.

May I ask @yangsophieee after installing all the dependencies did the vignettes get built successfully?

**Comment by dfalster** [comment]
Hi @fontikar 

You're welcome to elevate `janitor` in the dependencies (this is the one required by `summarise_austraits`), but not the others, especially not `raster`. It's a big install, requiring `terra` and `Rcpp`, both which build from source and have complex dependencies. We want to keep `austraits` relatively lightweight for basic install. 

I started wondering how other packages solve this and note that both `terra` nor `raster` have several packages in `suggests`, but don't have vignettes! (Hence they don't fail on install, when those decencies are missing). But then `terra` does have a package down site, just without "articles". 

I suggest scanning a few more packages to see how they handle this tension between suggests packages and vignettes. Maybe look at some depending on big stuff like raster

**Comment by yangsophieee** [comment]
>

**Comment by yangsophieee** [comment]
No worries @fontikar! Ah okay if I need to install it next time again, I'll use `dependencies = TRUE`! And yes the vignettes built successfully at the end after installing all dependencies!

**Comment by fontikar** [comment]
Brainstormed solutions, in order to keep the package as light as possible and vignettes to build, I will pre-render the vignettes using the file_name.Rmd.orig trick https://ropensci.org/blog/2019/12/08/precompute-vignettes/

**Comment by fontikar** [comment]
In the latest patch release, we've precompiled all vignettes so they don't require on the fly building using dependencies. This should streamline the installation process and avoid hiccups addressed in this issue. We've also included more detailed instructions in the README for installation, by setting `dependencies = TRUE` and `upgrade = ask` it forces the user install all dependencies (Imports, depends, suggests) to actively update any dependencies they have already installed.
